comp/otelcol/otlp/components/exporter/logsagentexporter/orchestrator_exporter.go

```diff
@@ -0,0 +1,554 @@
+			Type:     agentmodel.TypeCollectorManifest,
+		}, Body: payload})
+	if err != nil {
+		return fmt.Errorf("failed to encode payload: %w", err)
+	}
+
+	// Create the request
+	req, err := http.NewRequestWithContext(ctx, "POST", endpoint, bytes.NewReader(encoded))
+	if err != nil {
+		return fmt.Errorf("failed to create request: %w", err)
+	}
+
+	// Set headers
+	req.Header.Set("Content-Type", "application/x-protobuf")
+	req.Header.Set("DD-API-KEY", apiKey)
+	req.Header.Set("X-Dd-Hostname", hostName)
+	req.Header.Set("X-DD-Agent-Timestamp", strconv.Itoa(int(time.Now().Unix())))
+	req.Header.Set("X-Dd-Orchestrator-ClusterID", clusterID)
+	req.Header.Set("DD-EVP-ORIGIN", "agent")
+	req.Header.Set("DD-EVP-ORIGIN-VERSION", agentVersion)
+
+	// Send the request
+	resp, err := httpClient.Do(req)
+	if err != nil {
+		return fmt.Errorf("failed to send request: %w", err)
+	}
+	defer resp.Body.Close()
+
+	if resp.StatusCode != http.StatusOK && resp.StatusCode != http.StatusAccepted {
+		return fmt.Errorf("orchestrator endpoint returned non-200 status: %d", resp.StatusCode)
+	}
+
+	return nil
+}
+
+func getManifestType(kind string) int {
+	if _, ok := k8sTypeMap[kind]; ok {
+		return k8sTypeMap[kind]
+	}
+
+	return int(orchestratormodel.K8sUnsetType)
+}
+
+// ToDo: add more common tags
+func buildCommonTags() []string {
+	return []string{
+		"otel_receiver:k8sobjectsreceiver",
+	}
+}
+
+// ToDo: add more meaningful tags
+func buildTags(resource pcommon.Resource, logRecord plog.LogRecord) []string {
+	tags := buildCommonTags()
+
+	// Add resource attributes as tags
+	resource.Attributes().Range(func(key string, value pcommon.Value) bool {
+		tags = append(tags, fmt.Sprintf("otel_%s:%s", strings.ReplaceAll(key, ".", "_"), value.AsString()))
+		return true
+	})
+
+	// Add log record attributes as tags
+	logRecord.Attributes().Range(func(key string, value pcommon.Value) bool {
+		tags = append(tags, fmt.Sprintf("otel_%s:%s", strings.ReplaceAll(key, ".", "_"), value.AsString()))
+		return true
+	})
+
+	return tags
+}
+
+// shouldSkipResourceKind returns true if the Kubernetes resource kind should be skipped
+// for security or data sensitivity reasons. This matches the behavior of the orchestrator
+// collector which skips secrets and configmaps as they can contain sensitive data.
+func shouldSkipResourceKind(kind string) bool {
+	switch strings.ToLower(kind) {
+	case "secret", "configmap":
+		return true
+	default:
+		return false
+	}
+}
+
+// getEndpoint builds the orchestrator manifest endpoint URL from the provided configuration.
+// This follows the same pattern as the forwarder: Domain + Route (see transaction.HTTPTransaction.GetTarget)
+func getEndpoint(config OrchestratorConfig) (string, string) {
+	// Use the same endpoint route as defined in the forwarder
+	manifestRoute := endpoints.OrchestratorManifestEndpoint.Route
+
+	domain := fmt.Sprintf("https://orchestrator.%s", config.Site)
+	fullURL := domain + manifestRoute
```

Comment:
This getEndpoint function only needs to be called once, there’s no need to calculate the URL every time.